<!DOCTYPE html>
<html lang="en">
  

  <head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width,minimum-scale=1">

  <title>Allele flip when merging genotype data</title>
  <meta name="description" content="">

  <link rel="canonical" href="https://cumc.github.io/gaow-wiki/gaow-wiki/content/compbio_tutorial/allele_qc.html">
  <link rel="alternate" type="application/rss+xml" title="Lab Wiki" href="https://cumc.github.io/gaow-wiki/gaow-wiki/feed.xml">

  <meta property="og:url"         content="https://cumc.github.io/gaow-wiki/gaow-wiki/content/compbio_tutorial/allele_qc.html" />
<meta property="og:type"        content="article" />
<meta property="og:title"       content="Allele flip when merging genotype data" />
<meta property="og:description" content="" />
<meta property="og:image"       content="" />


  <script type="application/ld+json">
  {
  "@context": "http://schema.org",
  "@type": "NewsArticle",
  "mainEntityOfPage":
    "https://cumc.github.io/gaow-wiki/gaow-wiki/content/compbio_tutorial/allele_qc.html",
  "headline":
    "Allele flip when merging genotype data",
  "datePublished":
    "2022-02-24T21:12:26-06:00",
  "dateModified":
    "2022-02-24T21:12:26-06:00",
  "description":
    "",
  "author": {
    "@type": "Person",
    "name": "Gao Wang"
  },
  "publisher": {
    "@type": "Organization",
    "name": "Data 100 at UC Berkeley",
    "logo": {
      "@type": "ImageObject",
      "url": "https://cumc.github.io/gaow-wiki/gaow-wiki",
      "width": 60,
      "height": 60
    }
  },
  "image": {
    "@type": "ImageObject",
    "url": "https://cumc.github.io/gaow-wiki/gaow-wiki",
    "height": 60,
    "width": 60
  }
}

  </script>
  <link rel="stylesheet" href="/gaow-wiki/assets/css/styles.css">
  <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/font-awesome/4.7.0/css/font-awesome.min.css ">
  <link rel="apple-touch-icon" sizes="57x57" href="/apple-touch-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/apple-touch-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/apple-touch-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/apple-touch-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/apple-touch-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/apple-touch-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/apple-touch-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/apple-touch-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon-180x180.png">

  <!-- <link rel="manifest" href="/manifest.json"> -->
  <!-- <link rel="mask-icon" href="/safari-pinned-tab.svg" color="#efae0a"> -->
  <meta name="msapplication-TileColor" content="#da532c">
  <meta name="msapplication-TileImage" content="/mstile-144x144.png">
  <meta name="theme-color" content="#233947">

  <!-- Favicon -->
  <link rel="shortcut icon" type="image/x-icon" href="/gaow-wiki/images/favicon.ico">

  <!-- MathJax Config -->
  <!-- Allow inline math using $ and automatically break long math lines -->
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
    tex2jax: {
        inlineMath: [ ['$','$'], ["\\(","\\)"] ],
        displayMath: [ ['$$','$$'], ["\\[","\\]"] ],
        processEscapes: true,
        processEnvironments: true
    },
    CommonHTML: {
        linebreaks: {
            automatic: true,
        },
    },
});
</script>
<script src='https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js?config=TeX-AMS_CHTML' async></script>

  <!-- DOM updating function -->
  <script>
const runWhenDOMLoaded = cb => {
  if (document.readyState != 'loading') {
    cb()
  } else if (document.addEventListener) {
    document.addEventListener('DOMContentLoaded', cb)
  } else {
    document.attachEvent('onreadystatechange', function() {
      if (document.readyState == 'complete') cb()
    })
  }
}

// Helper function to init things quickly
initFunction = function(myfunc) {
  runWhenDOMLoaded(myfunc);
  document.addEventListener('turbolinks:load', myfunc);
};
</script>

  <!-- Define some javascript variables that will be useful in other javascript -->
  <script>
    const site_basename = '/gaow-wiki';
  </script>

  <!-- Add AnchorJS to let headers be linked -->
  <script src="/gaow-wiki/assets/js/anchor.min.js"  type="text/javascript"></script>
  <script>

initFunction(function () {
    anchors.add("main h1, main h2, main h3, main h4")
});

</script>

  <!-- Include Turbolinks to make page loads fast -->
  <!-- https://github.com/turbolinks/turbolinks -->
  <script src="/gaow-wiki/assets/js/turbolinks.js" async></script>
  <meta name="turbolinks-cache-control" content="no-cache">

  <!-- Load nbinteract for widgets -->
  

  <!-- Load Thebelab for interactive widgets -->
  <!-- Include Thebelab for interactive code if it's enabled -->



  <!-- Load the auto-generating TOC -->
  <script src="/gaow-wiki/assets/js/tocbot.min.js"  type="text/javascript"></script>
  <script>
var initToc = function () {
  tocbot.init({
    tocSelector: 'nav.onthispage',
    contentSelector: '.c-textbook__content',
    headingSelector: 'h2, h3',
    orderedList: false,
    collapseDepth: 6,
    listClass: 'toc__menu',
    activeListItemClass: "",  // Not using
    activeLinkClass: "", // Not using
  });
  tocbot.refresh();
}
initFunction(initToc);
</script>

  <!-- Google analytics -->
  <script src="/gaow-wiki/assets/js/ga.js" async></script>

  <!-- Clipboard copy button -->
  <script src="https://cdn.jsdelivr.net/npm/clipboard@2/dist/clipboard.min.js" async></script>

  <!-- Load JS that depends on site variables -->
  <script>
/**
 * Set up copy/paste for code blocks
 */
const codeCellId = index => `codecell${index}`

const clipboardButton = id =>
  `<a id="copy-button-${id}" class="btn copybtn o-tooltip--left" data-tooltip="Copy" data-clipboard-target="#${id}">
    <img src="/gaow-wiki/assets/images/copy-button.svg" alt="Copy to clipboard">
  </a>`

// Clears selected text since ClipboardJS will select the text when copying
const clearSelection = () => {
  if (window.getSelection) {
    window.getSelection().removeAllRanges()
  } else if (document.selection) {
    document.selection.empty()
  }
}

// Changes tooltip text for two seconds, then changes it back
const temporarilyChangeTooltip = (el, newText) => {
  const oldText = el.getAttribute('data-tooltip')
  el.setAttribute('data-tooltip', newText)
  setTimeout(() => el.setAttribute('data-tooltip', oldText), 2000)
}

const addCopyButtonToCodeCells = () => {
  // If ClipboardJS hasn't loaded, wait a bit and try again. This
  // happens because we load ClipboardJS asynchronously.
  if (window.ClipboardJS === undefined) {
    setTimeout(addCopyButtonToCodeCells, 250)
    return
  }

  const codeCells = document.querySelectorAll('div.c-textbook__content > div.highlighter-rouge > div.highlight > pre, div.input_area pre')
  codeCells.forEach((codeCell, index) => {
    const id = codeCellId(index)
    codeCell.setAttribute('id', id)
    if (document.getElementById("copy-button" + id) == null) {
      codeCell.insertAdjacentHTML('afterend', clipboardButton(id));
    }
  })

  const clipboard = new ClipboardJS('.copybtn')
  clipboard.on('success', event => {
    clearSelection()
    temporarilyChangeTooltip(event.trigger, 'Copied!')
  })

  clipboard.on('error', event => {
    temporarilyChangeTooltip(event.trigger, 'Failed to copy')
  })

  // Get rid of clipboard before the next page visit to avoid memory leak
  document.addEventListener('turbolinks:before-visit', () =>
    clipboard.destroy()
  )
}

initFunction(addCopyButtonToCodeCells);
</script>


  <!-- Hide cell code -->
  <script>
    /**
    Add buttons to hide code cells
    */


    var setCodeCellVisibility = function (inputField, kind) {
        // Update the image and class for hidden
        var id = inputField.getAttribute('data-id');
        var codeCell = document.querySelector(`#${id} div.highlight`);

        if (kind === "visible") {
            codeCell.classList.remove('hidden');
            inputField.checked = true;
        } else {
            codeCell.classList.add('hidden');
            inputField.checked = false;
        }
    }

    var toggleCodeCellVisibility = function (event) {
        // The label is clicked, and now we decide what to do based on the input field's clicked status
        if (event.target.tagName === "LABEL") {
            var inputField = event.target.previousElementSibling;
        } else {
            // It is the span inside the target
            var inputField = event.target.parentElement.previousElementSibling;
        }

        if (inputField.checked === true) {
            setCodeCellVisibility(inputField, "visible");
        } else {
            setCodeCellVisibility(inputField, "hidden");
        }
    }


    // Button constructor
    const hideCodeButton = id => `<input class="hidebtn" type="checkbox" id="hidebtn${id}" data-id="${id}"><label title="Toggle cell" for="hidebtn${id}" class="plusminus"><span class="pm_h"></span><span class="pm_v"></span></label>`

    var addHideButton = function () {
        // If a hide button is already added, don't add another
        if (document.querySelector('div.hidecode input') !== null) {
            return;
        }

        // Find the input cells and add a hide button
        document.querySelectorAll('div.input_area').forEach(function (item, index) {
            if (!item.classList.contains("hidecode")) {
                // Skip the cell if it doesn't have a hidecode class
                return;
            }

            const id = codeCellId(index)
            item.setAttribute('id', id);
            // Insert the button just inside the end of the next div
            item.querySelector('div').insertAdjacentHTML('beforeend', hideCodeButton(id))

            // Set up the visibility toggle
            hideLink = document.querySelector(`#${id} div.highlight + input + label`);
            hideLink.addEventListener('click', toggleCodeCellVisibility)
        });
    }


    // Initialize the hide buttos
    var initHiddenCells = function () {
        // Add hide buttons to the cells
        addHideButton();

        // Toggle the code cells that should be hidden
        document.querySelectorAll('div.hidecode input').forEach(function (item) {
            setCodeCellVisibility(item, 'hidden');
            item.checked = true;
        })
    }

    initFunction(initHiddenCells);

</script>

  <!-- Load custom website scripts -->
  <script src="/gaow-wiki/assets/js/scripts.js" async></script>

  <!-- Load custom user CSS and JS  -->
  <script src="/gaow-wiki/assets/custom/custom.js" async></script>
  <link rel="stylesheet" href="/gaow-wiki/assets/custom/custom.css">

  <!-- Update interact links w/ REST param, is defined in includes so we can use templates -->
  

  <!-- Lunr search code - will only be executed on the /search page -->
  <script src="/gaow-wiki/assets/js/lunr/lunr.min.js" type="text/javascript"></script>
  <script>var initQuery = function() {
  // See if we have a search box
  var searchInput = document.querySelector('input#lunr_search');
  if (searchInput === null) {
    return;
  }

  // Function to parse our lunr cache
  var idx = lunr(function () {
    this.field('title')
    this.field('excerpt')
    this.field('categories')
    this.field('tags')
    this.ref('id')

    this.pipeline.remove(lunr.trimmer)

    for (var item in store) {
      this.add({
        title: store[item].title,
        excerpt: store[item].excerpt,
        categories: store[item].categories,
        tags: store[item].tags,
        id: item
      })
    }
  });

  // Run search upon keyup
  searchInput.addEventListener('keyup', function () {
    var resultdiv = document.querySelector('#results');
    var query = document.querySelector("input#lunr_search").value.toLowerCase();
    var result =
      idx.query(function (q) {
        query.split(lunr.tokenizer.separator).forEach(function (term) {
          q.term(term, { boost: 100 })
          if(query.lastIndexOf(" ") != query.length-1){
            q.term(term, {  usePipeline: false, wildcard: lunr.Query.wildcard.TRAILING, boost: 10 })
          }
          if (term != ""){
            q.term(term, {  usePipeline: false, editDistance: 1, boost: 1 })
          }
        })
      });

      // Empty the results div
      while (resultdiv.firstChild) {
        resultdiv.removeChild(resultdiv.firstChild);
      }

    resultdiv.insertAdjacentHTML('afterbegin', '<p class="results__found">'+result.length+' Result(s) found</p>');
    for (var item in result) {
      var ref = result[item].ref;
      if(store[ref].teaser){
        var searchitem =
          '<div class="list__item">'+
            '<article class="archive__item" itemscope itemtype="https://schema.org/CreativeWork">'+
              '<h2 class="archive__item-title" itemprop="headline">'+
                '<a href="'+store[ref].url+'" rel="permalink">'+store[ref].title+'</a>'+
              '</h2>'+
              '<div class="archive__item-teaser">'+
                '<img src="'+store[ref].teaser+'" alt="">'+
              '</div>'+
              '<p class="archive__item-excerpt" itemprop="description">'+store[ref].excerpt.split(" ").splice(0,20).join(" ")+'...</p>'+
            '</article>'+
          '</div>';
      }
      else{
    	  var searchitem =
          '<div class="list__item">'+
            '<article class="archive__item" itemscope itemtype="https://schema.org/CreativeWork">'+
              '<h2 class="archive__item-title" itemprop="headline">'+
                '<a href="'+store[ref].url+'" rel="permalink">'+store[ref].title+'</a>'+
              '</h2>'+
              '<p class="archive__item-excerpt" itemprop="description">'+store[ref].excerpt.split(" ").splice(0,20).join(" ")+'...</p>'+
            '</article>'+
          '</div>';
      }
      resultdiv.insertAdjacentHTML('beforeend', searchitem);
    }
  });
};

initFunction(initQuery);
</script>
</head>

  <body>
    <!-- .js-show-sidebar shows sidebar by default -->
    <div id="js-textbook" class="c-textbook js-show-sidebar">
      



<nav id="js-sidebar" class="c-textbook__sidebar">
  <a href="http://wanggroup.org"><img src="/gaow-wiki/images/logo.png" class="textbook_logo" id="sidebar-logo" data-turbolinks-permanent/></a>
  <h2 class="c-sidebar__title">Lab Wiki</h2>
  <ul class="c-sidebar__chapters">
    
      
      
        <li class="c-sidebar__chapter"><a class="c-sidebar__entry" href="/gaow-wiki/search.html"><img src="/gaow-wiki/images/search.svg" alt="Search The Site"></a></li>
        
      
      

      
      

      
      
      <li class="c-sidebar__chapter">
        <a class="c-sidebar__entry "
          href="/gaow-wiki/index.html"
        >
          
          Home
        </a>

        
      </li>

      
    
      
      

      
      

      
      
      <li class="c-sidebar__chapter">
        <a class="c-sidebar__entry "
          href="/gaow-wiki/orientation/index.html"
        >
          
          Orientation
        </a>

        

          
          
          
          

          

          <ul class="c-sidebar__sections ">
            
              
              

              <li class="c-sidebar__section">
                <a class="c-sidebar__entry "
                  href="/gaow-wiki/orientation/5m-git.html"
                >
                  
                  5 minutes git
                </a>

                
                

              </li>
              
            
              
              

              <li class="c-sidebar__section">
                <a class="c-sidebar__entry "
                  href="/gaow-wiki/orientation/github-issues.html"
                >
                  
                  Use github issues
                </a>

                
                

              </li>
              
            
              
              

              <li class="c-sidebar__section">
                <a class="c-sidebar__entry "
                  href="/gaow-wiki/orientation/jupyter-setup.html"
                >
                  
                  Jupyter and SoS suite setup
                </a>

                
                

              </li>
              
            
          </ul>
        
      </li>

      
    
      
      

      
      

      
      
      <li class="c-sidebar__chapter">
        <a class="c-sidebar__entry "
          href="/gaow-wiki/project_resource/index.html"
        >
          
          Projects
        </a>

        

          
          
          
          

          

          <ul class="c-sidebar__sections ">
            
              
              

              <li class="c-sidebar__section">
                <a class="c-sidebar__entry "
                  href="/gaow-wiki/project_resource/project.html"
                >
                  
                  Project repositories
                </a>

                
                

              </li>
              
            
              
              

              <li class="c-sidebar__section">
                <a class="c-sidebar__entry "
                  href="/gaow-wiki/project_resource/pipeline.html"
                >
                  
                  Pipelines & benchmarks
                </a>

                
                

              </li>
              
            
              
              

              <li class="c-sidebar__section">
                <a class="c-sidebar__entry "
                  href="/gaow-wiki/project_resource/data.html"
                >
                  
                  Data resources
                </a>

                
                

              </li>
              
            
              
              

              <li class="c-sidebar__section">
                <a class="c-sidebar__entry "
                  href="/gaow-wiki/project_resource/sharing-guideline.html"
                >
                  
                  Sharing guidelines
                </a>

                
                

              </li>
              
            
          </ul>
        
      </li>

      
    
      
      
        <li class="c-sidebar__divider"></li>
        
      
      
        <li><h2 class="c-sidebar__title">Resources</li>
        
      
      

      
      

      
      
      <li class="c-sidebar__chapter">
        <a class="c-sidebar__entry "
          href="/gaow-wiki/learning_center/index.html"
        >
          
          Learning resource
        </a>

        
      </li>

      
    
      
      

      
      

      
      
      <li class="c-sidebar__chapter">
        <a class="c-sidebar__entry "
          href="/gaow-wiki/research_resource/index.html"
        >
          
          Research resource
        </a>

        
      </li>

      
    
      
      

      
      

      
      
      <li class="c-sidebar__chapter">
        <a class="c-sidebar__entry "
          href="/gaow-wiki/productivity_tips/index.html"
        >
          
          Productivity tips
        </a>

        
      </li>

      
    
      
      

      
      

      
      
      <li class="c-sidebar__chapter">
        <a class="c-sidebar__entry "
          href="/gaow-wiki/this_that/index.html"
        >
          
          This and that
        </a>

        
      </li>

      
    
      
      
        <li class="c-sidebar__divider"></li>
        
      
      
        <li><h2 class="c-sidebar__title">Lab tutorials</li>
        
      
      

      
      

      
      
      <li class="c-sidebar__chapter">
        <a class="c-sidebar__entry "
          href="/gaow-wiki/compbio_tutorial/index.html"
        >
          
          Computational biology
        </a>

        
      </li>

      
    
      
      

      
      

      
      
      <li class="c-sidebar__chapter">
        <a class="c-sidebar__entry "
          href="/gaow-wiki/computing_tutorial/index.html"
        >
          
          Computing
        </a>

        
      </li>

      
    
  </ul>
  <p class="sidebar_footer">Copyright (c) 2019+ <br> <a href="http://tigerwang.org">Gao Wang</a> and other lab members.<br>Distributed under <a href="https://creativecommons.org/licenses/by-sa/4.0/">CC BY-SA license</a>.<br></p>
</nav>

      
      <!-- Empty sidebar placeholder that we'll auto-fill with javascript -->
      <aside class="sidebar__right">
          <header><h4 class="nav__title"><i class="fa fa-list"></i>   Contents</h4></header>
          <nav class="onthispage">
          </nav>
      </aside>
      
      <main class="c-textbook__page" tabindex="-1">
          <div class="o-wrapper">
            <div class="c-sidebar-toggle">
  <!-- We show the sidebar by default so we use .is-active -->
  <button
    id="js-sidebar-toggle"
    class="hamburger hamburger--arrowalt is-active"
  >
    <span class="hamburger-box">
      <span class="hamburger-inner"></span>
    </span>
    <span class="c-sidebar-toggle__label">Toggle Sidebar</span>
  </button>
</div>

            

            <div class="c-textbook__content">
              <h1 id="allele-flip-when-merging-genotype-data">Allele flip when merging genotype data</h1>

<h2 id="problem">Problem</h2>

<p>Suppose we have some genetic variants measured in two sets of individuals and we would like to merge these genotype data.
Ideally, a variant, given by <code class="language-plaintext highlighter-rouge">chrom</code>, <code class="language-plaintext highlighter-rouge">position</code>, <code class="language-plaintext highlighter-rouge">reference allele</code>, <code class="language-plaintext highlighter-rouge">alternative allele</code>, would easily merge if information between
two data-sets are consistent. That is, the variant has the same 4 quantities for the different data-sets. However in reality there are
a few problems:</p>

<ul>
  <li>
    <p><strong><em>Scenario 1:</em></strong> When genotype data are given in PLINK format, there is no information on which allele is the reference and which is alternative allele.
 Instead all you get is “major” and “minor” allele, which is determined by the samples in the data-set. For example suppose a variant is:</p>

    <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code> 1:124235:A:G # chrom:pos:ref:alt
</code></pre></div>    </div>

    <p>and in the first data-set 60% alleles are <code class="language-plaintext highlighter-rouge">A</code>, whereas in the 2nd data-set only 45% are <code class="language-plaintext highlighter-rouge">A</code>. Then in the PLINK <code class="language-plaintext highlighter-rouge">bim</code> file the major allele will be
 A for data-set 1, and G for data-set 2. Even though in this case the two variants will appear</p>

    <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code> 1:124235:A:G # chrom:pos:major:minor
</code></pre></div>    </div>

    <p>and</p>

    <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code> 1:124235:G:A # chrom:pos:major:minor
</code></pre></div>    </div>

    <p>in different data-sets, they are still the same variant. Software such as PLINK will handle these major/minor allele inconsistency by internally flipping genotype coding
 when the variants are merged,
 as long as we make the variant ID the same for the two data-sets in PLINK <code class="language-plaintext highlighter-rouge">bim</code> file (via some simple programming) so PLINK knows they are the same variant.</p>
  </li>
  <li>
    <p><strong><em>Scenario 2:</em></strong> Sometimes a variant, measured on different strand due to differences in genotyping platform, can have different allele representations. For example in the first data-set:</p>

    <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code> 1:124235:A:G
</code></pre></div>    </div>

    <p>and the second:</p>

    <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code> 1:124235:T:C
</code></pre></div>    </div>

    <p>but in fact these two variants are the same because alleles A/G on their complementary DNA strand is <code class="language-plaintext highlighter-rouge">T/C</code>. In practise what we do is to change the allele representation in the 2nd data to</p>

    <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code> 1:124235:A:G
</code></pre></div>    </div>

    <p>to match and be ready to merge with the first data-set. This is called strand flip.</p>

    <p>Notice that the 2nd data-set may have <code class="language-plaintext highlighter-rouge">1:124235:C:T</code> which flips to <code class="language-plaintext highlighter-rouge">1:124235:G:A</code>, and falls in scenario #1 the major minor allele inconsistency which is straightforward to iron out with tools like PLINK.</p>
  </li>
  <li>
    <p><strong><em>Scenario 3:</em></strong>  Complication arise when it comes to specificly A/T and C/G genotypes. Suppose a variant in the first data-set:</p>

    <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code> 1:124236:A:T
</code></pre></div>    </div>

    <p>and the second:</p>

    <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code> 1:124236:T:A
</code></pre></div>    </div>

    <p>If it is a strand flip, the 2nd data-set on the complementary strand is</p>

    <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code> 1:124236:A:T
</code></pre></div>    </div>

    <p>which can be directly merged with the first data-set. However, it can also be a major/minor allele inconsistency in which case we do not flip the strand but will rely on PLINK to flip the allele coding while it merges the data.
 <strong>Since we cannot tell one situation from another, and the genotypes in these two situations will be handled differently</strong>, in practice we simply remove variants with A/T alleles from consideration.</p>

    <p>Same holds for C/G — you can work that out yourself to verify</p>
  </li>
  <li>
    <p>Tri-allelic case, such as one data-set has <code class="language-plaintext highlighter-rouge">A:C</code> the other has <code class="language-plaintext highlighter-rouge">A:G</code>, or <code class="language-plaintext highlighter-rouge">C:G</code> etc.</p>
  </li>
</ul>

<h2 id="implementation-of-a-solution">Implementation of a solution</h2>

<p>The R code below is extracted from <a href="http://gusevlab.org/projects/fusion/">FUSION</a> script by Hao Sun in our group:</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">allele.qc</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="k">function</span><span class="p">(</span><span class="n">a1</span><span class="p">,</span><span class="n">a2</span><span class="p">,</span><span class="n">ref1</span><span class="p">,</span><span class="n">ref2</span><span class="p">)</span><span class="w"> </span><span class="p">{</span><span class="w">
	</span><span class="c1"># a1 and a2 are the first data-set</span><span class="w">
	</span><span class="c1"># ref1 and ref2 are the 2nd data-set</span><span class="w">
	</span><span class="c1"># Make all the alleles into upper-case, as A,T,C,G:</span><span class="w">
        </span><span class="n">a1</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">toupper</span><span class="p">(</span><span class="n">a1</span><span class="p">)</span><span class="w">
        </span><span class="n">a2</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">toupper</span><span class="p">(</span><span class="n">a2</span><span class="p">)</span><span class="w">
        </span><span class="n">ref1</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">toupper</span><span class="p">(</span><span class="n">ref1</span><span class="p">)</span><span class="w">
        </span><span class="n">ref2</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">toupper</span><span class="p">(</span><span class="n">ref2</span><span class="p">)</span><span class="w">
	</span><span class="c1"># Strand flip, to change the allele representation in the 2nd data-set</span><span class="w">
	</span><span class="n">strand_flip</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="k">function</span><span class="p">(</span><span class="n">ref</span><span class="p">)</span><span class="w"> </span><span class="p">{</span><span class="w">
		</span><span class="n">flip</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">ref</span><span class="w">
		</span><span class="n">flip</span><span class="p">[</span><span class="n">ref</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s2">"A"</span><span class="p">]</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"T"</span><span class="w">
		</span><span class="n">flip</span><span class="p">[</span><span class="n">ref</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s2">"T"</span><span class="p">]</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"A"</span><span class="w">
		</span><span class="n">flip</span><span class="p">[</span><span class="n">ref</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s2">"G"</span><span class="p">]</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"C"</span><span class="w">
		</span><span class="n">flip</span><span class="p">[</span><span class="n">ref</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s2">"C"</span><span class="p">]</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"G"</span><span class="w">
		</span><span class="n">flip</span><span class="w">
	</span><span class="p">}</span><span class="w">
	</span><span class="n">flip1</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">strand_flip</span><span class="p">(</span><span class="n">ref1</span><span class="p">)</span><span class="w">
	</span><span class="n">flip2</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">strand_flip</span><span class="p">(</span><span class="n">ref2</span><span class="p">)</span><span class="w">
	</span><span class="n">snp</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">list</span><span class="p">()</span><span class="w">
	</span><span class="c1"># Remove strand ambiguous SNPs (scenario 3)</span><span class="w">
	</span><span class="n">snp</span><span class="p">[[</span><span class="s2">"keep"</span><span class="p">]]</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="o">!</span><span class="p">((</span><span class="n">a1</span><span class="o">==</span><span class="s2">"A"</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">a2</span><span class="o">==</span><span class="s2">"T"</span><span class="p">)</span><span class="w"> </span><span class="o">|</span><span class="w"> </span><span class="p">(</span><span class="n">a1</span><span class="o">==</span><span class="s2">"T"</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">a2</span><span class="o">==</span><span class="s2">"A"</span><span class="p">)</span><span class="w"> </span><span class="o">|</span><span class="w"> </span><span class="p">(</span><span class="n">a1</span><span class="o">==</span><span class="s2">"C"</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">a2</span><span class="o">==</span><span class="s2">"G"</span><span class="p">)</span><span class="w"> </span><span class="o">|</span><span class="w"> </span><span class="p">(</span><span class="n">a1</span><span class="o">==</span><span class="s2">"G"</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">a2</span><span class="o">==</span><span class="s2">"C"</span><span class="p">))</span><span class="w">
	</span><span class="c1"># Remove non-ATCG coding</span><span class="w">
	</span><span class="n">snp</span><span class="p">[[</span><span class="s2">"keep"</span><span class="p">]][</span><span class="w"> </span><span class="n">a1</span><span class="w"> </span><span class="o">!=</span><span class="w"> </span><span class="s2">"A"</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">a1</span><span class="w"> </span><span class="o">!=</span><span class="w"> </span><span class="s2">"T"</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">a1</span><span class="w"> </span><span class="o">!=</span><span class="w"> </span><span class="s2">"G"</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">a1</span><span class="w"> </span><span class="o">!=</span><span class="w"> </span><span class="s2">"C"</span><span class="w"> </span><span class="p">]</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nb">F</span><span class="w">
	</span><span class="n">snp</span><span class="p">[[</span><span class="s2">"keep"</span><span class="p">]][</span><span class="w"> </span><span class="n">a2</span><span class="w"> </span><span class="o">!=</span><span class="w"> </span><span class="s2">"A"</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">a2</span><span class="w"> </span><span class="o">!=</span><span class="w"> </span><span class="s2">"T"</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">a2</span><span class="w"> </span><span class="o">!=</span><span class="w"> </span><span class="s2">"G"</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">a2</span><span class="w"> </span><span class="o">!=</span><span class="w"> </span><span class="s2">"C"</span><span class="w"> </span><span class="p">]</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nb">F</span><span class="w">
	</span><span class="c1"># as long as scenario 1 is involved, sign_flip will return TRUE</span><span class="w">
	</span><span class="n">snp</span><span class="p">[[</span><span class="s2">"sign_flip"</span><span class="p">]]</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="p">(</span><span class="n">a1</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">ref2</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">a2</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">ref1</span><span class="p">)</span><span class="w"> </span><span class="o">|</span><span class="w"> </span><span class="p">(</span><span class="n">a1</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">flip2</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">a2</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">flip1</span><span class="p">)</span><span class="w">
	</span><span class="c1"># as long as scenario 2 is involved, strand_flip will return TRUE</span><span class="w">
	</span><span class="n">snp</span><span class="p">[[</span><span class="s2">"strand_flip"</span><span class="p">]]</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="p">(</span><span class="n">a1</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">flip1</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">a2</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">flip2</span><span class="p">)</span><span class="w"> </span><span class="o">|</span><span class="w"> </span><span class="p">(</span><span class="n">a1</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">flip2</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">a2</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">flip1</span><span class="p">)</span><span class="w">
	</span><span class="c1"># remove other cases, eg, tri-allelic, one dataset is A C, the other is A G, for example.</span><span class="w">
	</span><span class="n">exact_match</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="p">(</span><span class="n">a1</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">ref1</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">a2</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">ref2</span><span class="p">)</span><span class="w"> 
	</span><span class="n">snp</span><span class="p">[[</span><span class="s2">"keep"</span><span class="p">]][</span><span class="o">!</span><span class="p">(</span><span class="n">exact_match</span><span class="w"> </span><span class="o">|</span><span class="w"> </span><span class="n">snp</span><span class="p">[[</span><span class="s2">"sign_flip"</span><span class="p">]]</span><span class="w"> </span><span class="o">|</span><span class="w"> </span><span class="n">snp</span><span class="p">[[</span><span class="s2">"strand_flip"</span><span class="p">]])]</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nb">F</span><span class="w">
	</span><span class="nf">return</span><span class="p">(</span><span class="n">snp</span><span class="p">)</span><span class="w">
</span><span class="p">}</span><span class="w">
</span></code></pre></div></div>

<p>Each input <code class="language-plaintext highlighter-rouge">a1</code>, <code class="language-plaintext highlighter-rouge">a2</code>, <code class="language-plaintext highlighter-rouge">ref1</code>, <code class="language-plaintext highlighter-rouge">ref2</code> is a list of characters of <code class="language-plaintext highlighter-rouge">{ATCG}</code>.</p>

<h3 id="example-usage-1-merge-summary-statistics-data-with-reference-genotypes">Example usage 1: merge summary statistics data with reference genotypes</h3>

<p>In this application, <code class="language-plaintext highlighter-rouge">strand_flip</code> does not matter and only <code class="language-plaintext highlighter-rouge">sign_flip</code> matters. <code class="language-plaintext highlighter-rouge">sign_flip</code> is major/minor allele flip, and matters to the sign of z-scores</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">qc</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">allele.qc</span><span class="p">(</span><span class="w"> </span><span class="n">sumstat</span><span class="o">$</span><span class="n">A1</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="n">sumstat</span><span class="o">$</span><span class="n">A2</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="n">genos</span><span class="o">$</span><span class="n">bim</span><span class="p">[,</span><span class="m">5</span><span class="p">]</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="n">genos</span><span class="o">$</span><span class="n">bim</span><span class="p">[,</span><span class="m">6</span><span class="p">]</span><span class="w"> </span><span class="p">)</span><span class="w">
</span><span class="c1"># Flip Z-scores for mismatching alleles</span><span class="w">
</span><span class="n">sumstat</span><span class="o">$</span><span class="n">Z</span><span class="p">[</span><span class="w"> </span><span class="n">qc</span><span class="o">$</span><span class="n">sign_flip</span><span class="w"> </span><span class="p">]</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">-1</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="n">sumstat</span><span class="o">$</span><span class="n">Z</span><span class="p">[</span><span class="w"> </span><span class="n">qc</span><span class="o">$</span><span class="n">sign_flip</span><span class="w"> </span><span class="p">]</span><span class="w">
</span><span class="n">sumstat</span><span class="o">$</span><span class="n">A1</span><span class="p">[</span><span class="w"> </span><span class="n">qc</span><span class="o">$</span><span class="n">sign_flip</span><span class="w"> </span><span class="p">]</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">genos</span><span class="o">$</span><span class="n">bim</span><span class="p">[</span><span class="n">qc</span><span class="o">$</span><span class="n">sign_flip</span><span class="p">,</span><span class="m">5</span><span class="p">]</span><span class="w">
</span><span class="n">sumstat</span><span class="o">$</span><span class="n">A2</span><span class="p">[</span><span class="w"> </span><span class="n">qc</span><span class="o">$</span><span class="n">sign_flip</span><span class="w"> </span><span class="p">]</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">genos</span><span class="o">$</span><span class="n">bim</span><span class="p">[</span><span class="n">qc</span><span class="o">$</span><span class="n">sign_flip</span><span class="p">,</span><span class="m">6</span><span class="p">]</span><span class="w">
</span><span class="c1"># Remove strand ambiguous SNPs (if any)</span><span class="w">
</span><span class="k">if</span><span class="w"> </span><span class="p">(</span><span class="w"> </span><span class="nf">sum</span><span class="p">(</span><span class="o">!</span><span class="n">qc</span><span class="o">$</span><span class="n">keep</span><span class="p">)</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">0</span><span class="w"> </span><span class="p">)</span><span class="w"> </span><span class="p">{</span><span class="w">
	</span><span class="n">genos</span><span class="o">$</span><span class="n">bim</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">genos</span><span class="o">$</span><span class="n">bim</span><span class="p">[</span><span class="n">qc</span><span class="o">$</span><span class="n">keep</span><span class="p">,]</span><span class="w">
	</span><span class="n">genos</span><span class="o">$</span><span class="n">bed</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">genos</span><span class="o">$</span><span class="n">bed</span><span class="p">[,</span><span class="n">qc</span><span class="o">$</span><span class="n">keep</span><span class="p">]</span><span class="w">
	</span><span class="n">sumstat</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">sumstat</span><span class="p">[</span><span class="n">qc</span><span class="o">$</span><span class="n">keep</span><span class="p">,]</span><span class="w">
</span><span class="p">}</span><span class="w">
</span></code></pre></div></div>

<h3 id="example-usage-2-merge-two-genotype-data-set">Example usage 2: merge two genotype data-set</h3>

<p>When two data-sets are merged with PLINK, <code class="language-plaintext highlighter-rouge">sign_flip</code> (the major/minor allele flips) does not matter because PLINK will handle it internally. Only <code class="language-plaintext highlighter-rouge">strand_flip</code> matters and we have to fix the alleles for those that need a strand flip.</p>

<p>Suppose we have 12 same variants in 2 data-sets:</p>

<p>Data-set one:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>1  22 rs9617549   0 10874444 C T
2  22 rs5747224   0 11122151 G A
3  22 rs2456393   0 11122417 G T
4  22 rs9617550   0 10874444 A G
5  22 rs5747225   0 11122151 T C
6  22 rs2456394   0 11122417 A C
7  22 rs199668908 0 11123683 A G
8  22 rs199674018 0 11125296 A T
9  22 rs538546296 0 15282439 C G
10 22 rs551794980 0 15283826 T A
11 22 rsxxxxxxxxx 0 15284000 A C
12 22 rsyyyyyyyyy 0 15285000 A C
</code></pre></div></div>
<p>Data-set two:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>1  22 rs9617549   0 10874444 T C
2  22 rs5747224   0 11122151 A G
3  22 rs2456393   0 11122417 T G
4  22 rs9617550   0 10874444 T C
5  22 rs5747225   0 11122151 A G
6  22 rs2456394   0 11122417 T G
7  22 rs199668908 0 11123683 C T
8  22 rs199674018 0 11125296 T A
9  22 rs538546296 0 15282439 G C
10 22 rs551794980 0 15283826 T A
11 22 rsxxxxxxxxx 0 15284000 A G
12 22 rsyyyyyyyyy 0 15285000 C G
</code></pre></div></div>

<p>We can tell the variants are the same by <code class="language-plaintext highlighter-rouge">chr_pos</code>. For example the first variant: is from chromosome 22, and the start position is 10874444. Sometimes the variant ID (e.g.rs9617549) is not provided.</p>

<p>For the 12 lines (examples) of the above 2 data-sets:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>	Line 1-3: Scenario 1
	Line 4-6: Scenario 2
	Line 7: The combination of scenario 1 and scenario 2.
			(Falls into scenario 1, after strand flip)
	Line 8-10: Scenario 3
	Line 11-12: Trialleic cases We need to merge these 2 datasets together by `chr_pos`, then apply `allele.qc` onto the merged data. 
</code></pre></div></div>

<p>The <code class="language-plaintext highlighter-rouge">data_merge</code> should look like this:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>	chr_pos	        V1.x	V2.x	V3.x	V4.x	V5.x	V6.x	V1.y	V2.y	V3.y	V4.y	V5.y	V6.y
1	22:rs9617549	22	rs9617549	0	10874444	C	     T	       22	rs9617549	0	10874444	T	C
2	22:rs5747224	22	rs5747224	0	11122151	G	     A	       22	rs5747224	0	11122151	A	G
3	22:rs2456393	22	rs2456393	0	11122417	G	     T	       22	rs2456393	0	11122417	T	G
4	22:rs9617550	22	rs9617549	0	10874444	A	     G	       22	rs9617549	0	10874444	T	C
5	22:rs5747225	22	rs5747224	0	11122151	T	     C	       22	rs5747224	0	11122151	A	G
6	22:rs2456394	22	rs2456393	0	11122417	A	     C	       22	rs2456393	0	11122417	T	G
7	22:rs199668908	22	rs199668908	0	11123683	A	     G     	   22	rs199668908	0	11123683	C	T
8	22:rs199674018	22	rs199674018	0	11125296	A	     T	       22	rs199674018	0	11125296	T	A
9	22:rs538546296	22	rs538546296	0	15282439	C     	 G	       22	rs538546296	0	15282439	G	C
10	22:rs551794980	22	rs551794980	0	15283826	T	     A	       22	rs551794980	0	15283826	T	A
11	22:rsxxxxxxxxx	22	rsxxxxxxxxx	0	15284000	A	     C	       22	rsxxxxxxxxx	0	15284000	A	G
12	22:rsyyyyyyyyy	22	rsyyyyyyyyy	0	15285000	A	     C	       22	rsyyyyyyyyy	0	15285000	C	G
</code></pre></div></div>

<p>And we then apply <code class="language-plaintext highlighter-rouge">allele.qc</code> as:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>qc = allele.qc(data_merge[,'V5.x'] , data_merge[,'V6.x'] , data_merge[,'V5.y'], data_merge[,'V6.y'])

</code></pre></div></div>
<p>The interface of this merger, and how it works in detail can be found <a href="https://github.com/gaow/alzheimers-family/blob/master/notebook/20210224_Allele_QC_MWE_Analysis.ipynb">Here</a>.</p>

<p><strong>The results of testing the 12 SNPs are as expected</strong></p>

<ol>
  <li>
    <p><code class="language-plaintext highlighter-rouge">keep</code> would return FALSE for scenario 3 and triallelic cases, i.e. the last 5 examples.</p>
  </li>
  <li>
    <p><code class="language-plaintext highlighter-rouge">sign_flip</code> identifies those cases where scenario 1 is involved, i.e. TRUE for the first 3 variantes (scenario 1 only), the 7th variantes (combination of scanrio 1 and scenario 2), and scenario 3, lines 8-10.</p>
  </li>
  <li>
    <p><code class="language-plaintext highlighter-rouge">strand_flip</code> identifies those cases where scenario 2 is involved, i.e. TRUE for scenario 2 only (lines 4-6), the 7th variantes (combination of scanrio 1 and scenario 2), and scenario 3, line 8-10.</p>
  </li>
</ol>


              <nav class="c-page__nav">
  

  
</nav>

            </div>
          </div>
        </div>
      </main>
    </div>

  </body>
</html>
